pkg/xero/base.go

```diff
@@ -32,6 +37,23 @@ func (m HTTPMethod) String() string {
 	return "unknown"
 }
 
+func getIdempotencyKey(tenantId string, entity string) string {
+	// Generate a ULID based on the current timestamp
+	now := time.Now().UTC()
+	// Generate a new entropy source:
+	entropy := ulid.Monotonic(rand.Reader, 0)
+	// Generate a ULID:
+	id, err := ulid.New(ulid.Timestamp(now), entropy)
+
+	// Oooh dear :(
+	if err != nil {
+		log.Fatalf("failed to generate ULID: %v", err)
```

Comment:
Won't this kill who ever is trying to use it? Maybe return error instead so upstream can handle the fatal?

Reply:
Yeh good shout.

Reply:
Amended ✅

Reply:
For the minute, if this function returns an error, I don't set the header.